src/lookup_service.cpp

```diff
@@ -101,6 +103,64 @@ LookupService::registerData(const DmtcpMessage &msg, const void *data)
   addKeyValue(msg.nsid, key, keyLen, val, valLen);
 }
 
+void
+LookupService::get64(jalib::JSocket &remote,
+                     const DmtcpMessage &msg)
+{
+  KeyValueMap64 &kvmap = _maps64[msg.kvdbId];
+
+  if (kvmap.find(msg.kvdb.key) == kvmap.end()) {
+    JTRACE("Lookup Failed, Key not found.");
+    remote << DmtcpMessage(DMT_KVDB64_GET_FAILED);
+  }
+
+  DmtcpMessage reply(DMT_KVDB64_GET_RESPONSE);
+  reply.kvdb.value = kvmap[msg.kvdb.key];
+
+  remote << reply;
+}
+
+void
+LookupService::set64(const DmtcpMessage &msg)
+{
+  KeyValueMap64 &kvmap = _maps64[msg.kvdbId];
+
+  if (msg.kvdb.op == DMTCP_KVDB_SET) {
+    kvmap[msg.kvdb.key] = msg.kvdb.value;
+    return;
+  }
+
+  switch (msg.kvdb.op)
+  {
+  case DMTCP_KVDB_INCRBY:
+    kvmap[msg.kvdb.key] += msg.kvdb.value;
+    break;
+
+  case DMTCP_KVDB_OR:
+    kvmap[msg.kvdb.key] |= msg.kvdb.value;
+    break;
+
+  case DMTCP_KVDB_XOR:
+    kvmap[msg.kvdb.key] ^= msg.kvdb.value;
+    break;
+
+  case DMTCP_KVDB_NOT:
+    kvmap[msg.kvdb.key] = ~kvmap[msg.kvdb.key];
+    break;
+
+  case DMTCP_KVDB_AND:
+    if (kvmap.find(msg.kvdb.key) == kvmap.end()) {
+      kvmap[msg.kvdb.key] = msg.kvdb.value;
+    }
+
+    kvmap[msg.kvdb.key] &= kvmap[msg.kvdb.key];
```

Comment:
Is the key's value supposed to AND to itself or `msg.kvdb.value`?
Also, could you add a comment stating why you assume the value to be present in other operations (OR and XOR) but not in AND? In line 153, you set the value if the key is not found, do you intend to use AND as a set operation?

Reply:
Fixed the bug and added a comment.